### PR TITLE
Priorities should be optional

### PR DESCRIPTION
**Pull request by nitrocode**
@aknysh @osterman please review if time permits

https://www.terraform.io/docs/providers/aws/r/lb_listener_rule.html

> `priority` - (Optional) The priority for the rule between `1` and `50000`. Leaving it unset will automatically set the rule with next available priority after currently existing highest rule. A listener can't have multiple rules with the same priority.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -51,7 +51,7 @@ resource "aws_lb_target_group" "default" {
 resource "aws_lb_listener_rule" "unauthenticated_paths" {
   count        = length(var.unauthenticated_paths) > 0 && length(var.unauthenticated_hosts) == 0 ? var.unauthenticated_listener_arns_count : 0
   listener_arn = var.unauthenticated_listener_arns[count.index]
-  priority     = var.unauthenticated_priority + count.index
+  priority     = var.unauthenticated_priority > 0 ? var.unauthenticated_priority + count.index : null
```
Comment (Gowiem) [inline main.tf:59]: @nitrocode Previously this module never passed `null` as a value to these resources. Have you tested out what passing `null` does? I'm wondering if it was cause inconsistent applies or possibly errors?

**Comment by nitrocode** [comment]
Ah same s3 bucket naming issue as mentioned in https://github.com/cloudposse/terraform-aws-alb-ingress/pull/27#issuecomment-645338064

**Review by Gowiem — CHANGES_REQUESTED** [review]

**Comment by nitrocode** [comment]
@Gowiem 

> Leaving it unset will automatically set the rule with next available priority after currently existing highest rule.

By setting it to `null`, the rule priority gets set automatically by amazon instead of us.

**Review by Gowiem — APPROVED** [review]
Aha @nitrocode, that sounds good 👍 :shipit:

**Comment by Gowiem** [comment]
@nitrocode Shipped as [0.11.0](https://github.com/cloudposse/terraform-aws-alb-ingress/releases/tag/0.11.0) 😎

**Pull request merged**